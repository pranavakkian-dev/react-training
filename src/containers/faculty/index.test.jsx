import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import userEvent from '@testing-library/user-event';
import Store from '../../redux/store';
import Faculty from './index';

describe('Faculty', () => {
  test('Component rendered successfully!', () => {
    const { container } = render(
      <Provider store={Store}>
        <Faculty />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  test('Faculty created and displayed successfully!', () => {
    render(
      <Provider store={Store}>
        <Faculty />
      </Provider>,
    );
    userEvent.type(screen.getByRole('spinbutton', { name: 'ID:' }), '101');
    userEvent.type(screen.getByRole('textbox', { name: 'Name:' }), 'user1');
    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
    userEvent.type(screen.getByRole('spinbutton', { name: 'ID:' }), '102');
    userEvent.type(screen.getByRole('textbox', { name: 'Name:' }), 'user2');
    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
    expect(screen.getByRole('cell', { name: '101' })).toBeInTheDocument();
    expect(screen.getByRole('cell', { name: 'user1' })).toBeInTheDocument();
    expect(screen.getByRole('cell', { name: '102' })).toBeInTheDocument();
    expect(screen.getByRole('cell', { name: 'user2' })).toBeInTheDocument();
  });

  test('Faculty deleted and removed from the table successfully!', () => {
    render(
      <Provider store={Store}>
        <Faculty />
      </Provider>,
    );
    userEvent.type(screen.getByRole('spinbutton', { name: 'ID:' }), '101');
    userEvent.type(screen.getByRole('textbox', { name: 'Name:' }), 'user1');
    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
    userEvent.click(screen.getByRole('button', { name: 'Delete' }));
    expect(screen.queryByRole('cell', { name: '101' })).not.toBeInTheDocument();
    expect(screen.queryByRole('cell', { name: 'user1' })).not.toBeInTheDocument();
  });

  test('Store reset on un-mount successfully!', () => {
    const { unmount } = render(
      <Provider store={Store}>
        <Faculty />
      </Provider>,
    );
    userEvent.type(screen.getByRole('spinbutton', { name: 'ID:' }), '101');
    userEvent.type(screen.getByRole('textbox', { name: 'Name:' }), 'user1');
    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
    expect(Store.getState().FacultyReducer.faculties).toHaveLength(1);
    unmount();
    expect(Store.getState().FacultyReducer.faculties).toHaveLength(0);
  });
});
