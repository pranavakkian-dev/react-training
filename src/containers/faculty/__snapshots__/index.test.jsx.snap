// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Faculty Component rendered successfully! 1`] = `
<div>
  <h4>
    Add Faculty
  </h4>
  <label
    for="facultyId"
  >
    ID:
    <br />
    <input
      id="facultyId"
      type="number"
      value=""
    />
  </label>
  <br />
  <label
    for="facultyName"
  >
    Name:
    <br />
    <input
      id="facultyName"
      value=""
    />
  </label>
  <br />
  <button
    type="button"
  >
    Submit
  </button>
  <br />
  <br />
  <table>
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Name
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
